test/val/val_id_test.cpp

```diff
@@ -2780,6 +2780,35 @@ TEST_F(ValidateIdWithMessage, OpStoreBitcastNonPointerBad) {
   EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
 }
 
+// Result <id> resulting from an instruction within a function may not be used
+// outside that function.
+TEST_F(ValidateIdWithMessage, ResultIdUsedOutsideOfFunctionBad) {
+  string spirv = kGLSL450MemoryModel + R"(
+%1 = OpTypeVoid
+%2 = OpTypeFunction %1
+%3 = OpTypeInt 32 0
+%4 = OpConstant %3 1234
```

Comment:
You can shorten this example so that you don't need the %4 (OpConstant) or %9 (OpIAdd).

Change the OpStore at line 2800 into an OpLoad.